lib/rspec/mocks/error_generator.rb

```diff
@@ -142,7 +142,7 @@ def expected_part_of_expectation_error(expected_received_count, expectation_coun
       def actual_method_call_args_description(count, args)
         method_call_args_description(args) ||
           if count > 0 && args.length > 0
-            " with arguments: #{args.inspect.gsub(/\A\[(.+)\]\z/, '(\1)')}"
+            " with arguments: #{format_args(*args)}"
           else
             ""
           end
```

Comment:
I'm a little hesitant about the last commit - but was the best way I could find of unifying these two methods while preserving the same behavior. Let me know what you think.  Thanks!